crypto/src/address/r1cs.rs

```diff
@@ -0,0 +1,76 @@
+use crate::Address;
+use ark_ff::PrimeField;
+use ark_r1cs_std::prelude::*;
+use ark_relations::r1cs::SynthesisError;
+use decaf377::{
+    r1cs::{ElementVar, FqVar},
+    Element, Fq,
+};
+
+pub struct AddressVar {
+    pub diversified_generator: ElementVar,
```

Comment:
When working in the circuit, do we want to be working with the diversified generator `g_d` or the diversifier `d`? Do we need to do the `d -> g_d` map in the circuit?